internal/app/grpcserver/server.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcserver
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"net"
+
+	"connectrpc.com/conformance/internal/codec"
+	"connectrpc.com/conformance/internal/config"
+	conformancev1alpha1 "connectrpc.com/conformance/internal/gen/proto/go/connectrpc/conformance/v1alpha1"
+	"google.golang.org/grpc"
+)
+
+// Run runs the server according to server config read from the 'in' reader.
+func Run(_ context.Context, _ []string, inReader io.ReadCloser, outWriter io.WriteCloser, _ io.WriteCloser) error {
+	json := flag.Bool("json", false, "whether to use the JSON format for marshaling / unmarshaling messages")
```

Comment:
See what I did in #626 -- instead of ignoring the `args` param (second param, `[]string` type), you need to [use it to parse flags](https://github.com/connectrpc/conformance/pull/626/files#diff-183b1c01db7b47b94d058df3d38c5f457214d8082087a04c107ffb50040e9a64R41-R46). Otherwise, when run in-process, this is actually looking at the flags provided to the test runner instead of the flags passed as argument to this function.